library/board.py

```diff
@@ -21,11 +21,24 @@ def __init__(self) -> None:
 
     def check_board_victory(self) -> typing.Optional[str]:
         """Determine if the entire board has been won, and by who."""
-        grid_states = np.array(
-            (self.check_subboard_victory(i) for i in "789456123"), dtype=np.str_
-        )
-        # This might have off ordering, TODO on testing it
-        grid_states.reshape((3, 3))
+        grid_states = np.full((3, 3), "·")
+        grid_map = {
+            1: [2, 0],
+            2: [2, 1],
+            3: [2, 2],
+            4: [1, 0],
+            5: [1, 1],
+            6: [1, 2],
+            7: [0, 0],
+            8: [0, 1],
+            9: [0, 2],
+        }
```

Comment:
Might be worth it to note that we are assuming the board is arranged like so
```
7 8 9
4 5 6
1 2 3
```

Though not sure whether this deserves to be in the docstring or in a comment above `grid_map`.

Reply:
Changed the docstring to this: 
```
"""
        Determine if the entire board has been won, and by who.

        This function creates a 3 x 3 array with the current winner of each subgrid.
        The 3 x 3 is arranged like this:

        7 8 9
        4 5 6
        1 2 3
        """
```